File: src/main/scala/fi/finlit/edith/sql/domain/Document.scala
package fi.finlit.edith.sql.domain

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table
import org.hibernate.annotations.Cache
import org.hibernate.annotations.CacheConcurrencyStrategy
import fi.finlit.edith.Identifiable
import scala.reflect.BeanProperty
//remove if not needed
import scala.collection.JavaConversions._

@Entity
@Table(name = "document")
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
class Document extends Identifiable {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  @BeanProperty
  var id: java.lang.Long = _

  @Column(unique = true)
  @BeanProperty
  var path: String = _

  @BeanProperty
  var title: String = _
}

File: src/main/java/com/mysema/edith/domain/Document.java
/*
 * Copyright (c) 2018 Mysema
 */
package com.mysema.edith.domain;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.io.File;

@Entity
@Table(name = "document")
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
public class Document extends BaseEntity {

    @Column(unique = true)
    private String path;

    private String title;

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public String getTitle() {
        File f = new File(this.getPath());
        return f.getName().toString();
      //  return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

}
